pkgs/development/libraries/fastcdr/default.nix

```
@@ -0,0 +1,63 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, gtest
+
+, withDocs ? true
+, doxygen
+, graphviz-nox
+}:
+
+let
+in stdenv.mkDerivation (finalAttrs: {
+  pname = "fastcdr";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "eProsima";
+    repo = "Fast-CDR";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-ZJQnm3JN56y2v/XIShfZxkEEu1AKMJxt8wpRqSn9HWk=";
+  };
+
+  patches = [
+    ./0001-Do-not-require-wget-and-unzip.patch
+    ./0002-Load-standard-install-dir-variables.patch
+  ];
+
+  cmakeFlags = lib.optional (stdenv.hostPlatform.isStatic) "-DBUILD_SHARED_LIBS=OFF"
+  # fastcdr doesn't respect BUILD_TESTING
+  ++ lib.optional (stdenv.hostPlatform == stdenv.buildPlatform) "-DEPROSIMA_BUILD_TESTS=ON"
+  ++ lib.optional withDocs "-DBUILD_DOCUMENTATION=ON";
+
+  outputs = [ "out" ] ++ lib.optional withDocs "doc";
+
+  nativeBuildInputs = [ cmake ]
+  ++ lib.optionals withDocs [
+    doxygen
+    graphviz-nox
+  ];
+
+  doCheck = true;
+
+  checkInputs = [ gtest ];
+
+  postInstall = lib.optionalString withDocs ''
+    mkdir -p $doc/share/doc/
+    mv $out/share/doc $doc/share/doc/${finalAttrs.pname}
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/eProsima/Fast-CDR";
+    description = ''
```

```suggestion
    longDescription = ''
```

And provide a short one.